### PR TITLE
Add mutation to update company fields

### PR DESCRIPTION
**Pull request by solocommand**
```gql
mutation UpdateContentCompany($company: UpdateContentCompanyMutationInput!){
  updateContentCompany(input:$company){
    id
    name
  }
}

mutation UpdateContentCompanyImages($images: UpdateContentCompanyImagesMutationInput!){
  updateContentCompanyImages(input:$images) {
    primaryImage {
      id
      src
    }
    images {
      edges {
        node {
          id
          src
        }
      }
    }
  }
}

mutation UpdateContentCompanySocialLinks($social: UpdateContentCompanySocialLinksMutationInput!){
  updateContentCompanySocialLinks(input:$social){
    id
    name
    socialLinks {
      provider
      url
    }
  }
}
```
```json
{
  "company": {
    "id": 30000001,
    "payload": {
    	"name":"TestCo 2"
    }
  },
  "images":{
    "id": 30000001,
    "payload":{
    	"primaryImage": "5deec6102931d62c008b4568",
    	"images":["5deec6102931d62c008b4568"]
    }
  },
  "social": {
    "id": 30000001,
    "payload":{
    	"socialLinks": [
     	 	{
      	  "provider":"facebook",
        	"url":"https://facebook.com/username"
      	}
    	]
    }
  }
}
```

### CODE REVIEW

File: services/graphql-server/src/graphql/definitions/platform/content/types/company.js [hunk 1]
```diff
@@ -6,6 +6,10 @@ extend type Query {
   contentCompany(input: ContentCompanyQueryInput!): ContentCompany @findOne(model: "platform.Content", using: { id: "_id" }, criteria: "contentCompany")
 }
 
+extend type Mutation {
+  contentCompanyUpdate(input: ContentCompanyMutationInput!): ContentCompany @requiresAuth
```
Comment (zarathustra323) [inline services/graphql-server/src/graphql/definitions/platform/content/types/company.js]: updateContentCompany please.

File: services/graphql-server/src/graphql/definitions/platform/content/types/company.js [hunk 2]
```diff
@@ -55,6 +59,40 @@ input ContentCompanyQueryInput {
...
+  zip: String
+  country: String
+  phone: String
+  tollfree: String
+  fax: String
+  website: String
+  type: String
+  email: String
+  body: String
+  socialLinks: [ContentCompanySocialLinkInput!]
```
Comment (zarathustra323) [inline services/graphql-server/src/graphql/definitions/platform/content/types/company.js]: I would handle social links separately.

File: services/graphql-server/src/graphql/resolvers/platform/content.js [hunk 3]
```diff
@@ -1119,4 +1121,24 @@ module.exports = {
...
+  /**
+   *
+   */
+  Mutation: {
+    contentCompanyUpdate: async (_, { input }, { base4rest, basedb }, info) => {
+      validateRest(base4rest);
+      const type = 'platform/content/company';
+      const { id, ...payload } = input;
+      const keys = Object.keys(payload);
+      if (!keys.length) throw new UserInputError('You must specify a field to update!');
```
Comment (zarathustra323) [inline services/graphql-server/src/graphql/resolvers/platform/content.js]: No need throw an error

File: services/graphql-server/src/graphql/resolvers/platform/content.js [hunk 4]
```diff
@@ -1119,4 +1121,24 @@ module.exports = {
       });
     },
   },
+
+  /**
+   *
+   */
+  Mutation: {
+    contentCompanyUpdate: async (_, { input }, { base4rest, basedb }, info) => {
+      validateRest(base4rest);
```
Comment (zarathustra323) [inline services/graphql-server/src/graphql/resolvers/platform/content.js]: Should require auth

Comment (solocommand) [reply services/graphql-server/src/graphql/resolvers/platform/content.js]: `@requiresAuth` is present on the [definition](https://github.com/base-cms/base-cms/pull/278/files#diff-bc933e054c1ded476ecb27848eaf79ffR10)

File: services/graphql-server/src/graphql/definitions/platform/content/types/company.js [hunk 5]
```diff
@@ -55,6 +61,50 @@ input ContentCompanyQueryInput {
   status: ModelStatus = active
 }
 
+input ContentCompanyMutationInput {
+  id: Int!
+  name: String
```
Comment (zarathustra323) [inline services/graphql-server/src/graphql/definitions/platform/content/types/company.js]: Set all of these as a `payload` child input

File: services/graphql-server/src/graphql/definitions/platform/content/types/company.js [hunk 6]
```diff
@@ -55,6 +61,50 @@ input ContentCompanyQueryInput {
...
+  servicesProvided: String
+  salesChannels: String
+  productSummary: String
+  serviceInformation: String
+  warrantyInformation: String
+}
+
+input ContentCompanyImagesMutationInput {
+  id: Int!
+  primaryImage: ObjectID
```
Comment (zarathustra323) [inline services/graphql-server/src/graphql/definitions/platform/content/types/company.js]: Set to `payload` child input object

File: services/graphql-server/src/graphql/definitions/platform/content/types/company.js [hunk 7]
```diff
@@ -55,6 +61,50 @@ input ContentCompanyQueryInput {
...
+  images: [ObjectID!]
+}
+
+input ContentCompanySocialLinksMutationInput {
+  id: Int!
+  socialLinks: [ContentCompanySocialLinkInput!]!
```
Comment (zarathustra323) [inline services/graphql-server/src/graphql/definitions/platform/content/types/company.js]: Payload child object

File: services/graphql-server/src/graphql/resolvers/platform/content.js [hunk 8]
```diff
@@ -1119,4 +1121,49 @@ module.exports = {
...
+    updateContentCompany: async (_, { input }, { base4rest, basedb }, info) => {
+      validateRest(base4rest);
+      const type = 'platform/content/company';
+      const { id, ...payload } = input;
+      const keys = Object.keys(payload);
+      const body = new Base4RestPayload({ type });
+      keys.forEach(k => body.set(k, payload[k]));
+      body.set('id', id);
+      await base4rest.updateOne({ model: type, id, body });
+      const { fieldNodes, schema, fragments } = info;
```
Comment (zarathustra323) [inline services/graphql-server/src/graphql/resolvers/platform/content.js]: Consolidate this projection handling... should ultimately defer to `findOne` directive.

File: services/graphql-server/src/graphql/resolvers/platform/content.js [hunk 9]
```diff
@@ -1119,4 +1121,49 @@ module.exports = {
...
   },
+
+  /**
+   *
+   */
+  Mutation: {
+    updateContentCompany: async (_, { input }, { base4rest, basedb }, info) => {
+      validateRest(base4rest);
+      const type = 'platform/content/company';
+      const { id, ...payload } = input;
```
Comment (zarathustra323) [inline services/graphql-server/src/graphql/resolvers/platform/content.js]: The `payload` value will need to be fixed.

**Pull request merged**